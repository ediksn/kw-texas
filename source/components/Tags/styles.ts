import { StyleSheet } from 'react-native'
import { scale, moderateScale, verticalScale } from 'react-native-size-matters'
import { theme } from '~/constants'

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: verticalScale(48),
    marginLeft: scale(16)
  },
  tags: {
    backgroundColor: '#D5F7FF',
    marginRight: scale(4),
    paddingVertical: moderateScale(4),
    paddingLeft: moderateScale(8),
    paddingRight: moderateScale(5),
    borderRadius: moderateScale(12),
    flexDirection: 'row',
    alignItems: 'center'
  },
  name: {
    fontFamily: 'Mulish-Regular',
    fontWeight: '700',
    fontSize: theme.fonts.SMALL_SIZE,
    lineHeight: 20,
    color: '#006C84',
    marginRight: scale(5)
  },
  closeIcon: {
    height: moderateScale(14),
    width: moderateScale(14)
  }
})
