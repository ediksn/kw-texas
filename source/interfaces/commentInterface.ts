export interface CommentDetailInterface {
  content: string
  edited: boolean
  likesCount: number
  userHasAlreadyLiked: boolean
  attachment: any
}

export interface CommentInterface {
  id: string
  creatorId: string
  createdAt: string
  type: string
  creatorPhoto: string
  creatorFirstName: string
  creatorLastName: string
  deleted: boolean
  postId: string
  updatedAt: string
  detail: CommentDetailInterface
}
