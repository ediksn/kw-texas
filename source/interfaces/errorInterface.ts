export interface ErrorInterface {
  error_description: string | null
  error: string | null
}
