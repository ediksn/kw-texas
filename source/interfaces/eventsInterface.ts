import moment from 'moment'

interface InstructorsInterface {
  firstName: string
  lastName: string
}

export interface EventsInterface {
  id: number
  name: string
  price: string
  location: string
  ends: moment.Moment
  starts: moment.Moment
  published: boolean
  instructors: InstructorsInterface[]
  bookmarked: boolean
  slug: any
}
