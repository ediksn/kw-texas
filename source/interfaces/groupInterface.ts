import { TextStyle, ViewStyle } from 'react-native'

export interface OptionInterface {
  key?: string
  isTitle?: boolean
  handleOption?: () => void
  title?: string
  color?: string
}

export interface GroupOptionInterface {
  visible: boolean
  buttonRef: any
  options: OptionInterface[]
  styleContainer?: ViewStyle
  styleOption?: TextStyle
  toRight?: boolean
  onRequestClose: () => void
}

interface Icon {
  url: string
  id: string
  filename: string
}

interface LocationInterface {
  city: string
  state: string
  postalCode: string
  country: string
}

interface MemberInterface {
  id: string
  kwUid: number
  username: string
  firstName: string
  lastName: string
  email: string
  memberType: string
  state: string
  phone: string
  photoUrl: string
  location: LocationInterface
}
export interface GroupInterface {
  id: string
  name: string
  description: string
  postCount: number
  membersCount: number
  status: number
  icon: Icon
  members: MemberInterface[]
}

export interface GroupProduceProps {
  type: string
  payload?: {
    data: GroupInterface[]
    limit: number
  }
}

export interface GroupReducerProps {
  draftState: any
  payload?: {
    data: GroupInterface[]
    limit: number
  }
}
