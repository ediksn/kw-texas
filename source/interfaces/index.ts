export { default as loginInterface } from './loginInterface'
export { default as promptVideoInterface } from './promptVideoInterface'
