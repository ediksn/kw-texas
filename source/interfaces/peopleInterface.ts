export interface PeopleMarkedInterface {
  [kwUid: number]: boolean
}

interface LocationInterface {
  country: string | null
  state: string
  city: string
  address: string | null
  postalCode: string | null
}
export interface PeopleInterface {
  kwUid: number
  firstName: string
  lastName: string
  location: LocationInterface
  photoUrl: string
}

export interface PeopleProduceProps {
  type: string
  payload?: {
    data: PeopleInterface[]
    limit: number
  }
}

export interface PeopleReducerProps {
  draftState: any
  payload?: any
}
