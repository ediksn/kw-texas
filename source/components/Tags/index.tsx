import React from 'react'
import { TouchableOpacity, View, Text, Image, ScrollView } from 'react-native'
import closeTagImage from 'assets/images/close-tag.png'
import { styles } from './styles'
import { PeopleInterface, PeopleMarkedInterface } from '~/interfaces/peopleInterface'
import { useGetFullNamePeopleTags } from '~/hooks/tags'

interface TagsProps {
  tags: PeopleMarkedInterface
  peopleList: PeopleInterface[]
  handlePeopleMarked: (kwUid: number) => void
}

const Tags = ({ tags, peopleList, handlePeopleMarked }: TagsProps) => (
  <View style={styles.container}>
    <ScrollView horizontal>
      {Object.keys(tags).map((kwUid: string) => {
        const kwUidNumber = parseInt(kwUid, 10)

        return (
          tags[kwUidNumber] && (
            <View key={kwUidNumber} style={styles.tags}>
              <Text style={styles.name}>{useGetFullNamePeopleTags(peopleList, kwUidNumber)}</Text>
              <TouchableOpacity onPress={() => handlePeopleMarked(kwUidNumber)}>
                <Image source={closeTagImage} style={styles.closeIcon} />
              </TouchableOpacity>
            </View>
          )
        )
      })}
    </ScrollView>
  </View>
)

export default Tags
