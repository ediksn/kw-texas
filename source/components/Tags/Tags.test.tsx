import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import Tags from './index'
import { PeopleMarkedInterface } from '~/interfaces/peopleInterface'

describe('Render People Component Test', () => {
  let component: RenderAPI
  const mockTags: PeopleMarkedInterface = {}

  beforeEach(() => {
    component = render(<Tags tags={mockTags} peopleList={[]} handlePeopleMarked={() => null} />)
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
  })
})
