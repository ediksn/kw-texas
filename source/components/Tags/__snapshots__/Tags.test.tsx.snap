// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render People Component Test Match with the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 94.16470588235293,
      "marginLeft": 34.285714285714285,
    }
  }
>
  <RCTScrollView
    horizontal={true}
  >
    <View />
  </RCTScrollView>
</View>
`;
